translate/src/context/TranslationContext.tsx

```diff
@@ -29,42 +34,49 @@ const LLMTranslationContext = createContext<LLMTranslationContextType>({
 });
 
 export const LLMTranslationProvider: React.FC = ({ children }) => {
-  const translation = useActiveTranslation();
-  const [state, setState] = useState(new Map<MachineryTranslation, SelState>());
+  const stateRef = useRef(new WeakMap<MachineryTranslation, SelState>());
+  const [version, setVersion] = useState(0); // Counter to trigger re-renders
 
-  useEffect(() => setState(new Map()), [translation]);
-
-  const getSelState = (mt: MachineryTranslation) =>
-    state.get(mt) ?? initSelState();
+  const getSelState = useCallback(
```

Comment:
Wrapping this in a callback is unnecessary, as the only time this component re-renders is when `version` changes.